### PR TITLE
Upgrade `structopt` to `clap`

### PR DESCRIPTION
**Pull request by max-sixty**
~Still need to finish this off, one test fails.~

`structopt` was build on clap 2, and then its features were integrated into clap 3 — ref https://github.com/TeXitoi/structopt/issues/525. So it's not actually a heavier dependency. May have to check on MSRV issues.

Once we merge this, we only have one version of `syn`, which reduces our dependency size significantly.

---

This is ready for any feedback / testing. Will need to adjust the dependency spec if this gets in before we increase the `cargo-insta` msrv to 1.64...

- It's stacked on https://github.com/mitsuhiko/insta/pull/517, so we should merge that first
- It requires #524 for CI to pass, at least without manually pinning dependencies

### CODE REVIEW

**Comment by mitsuhiko** [comment]
I'm in favor. I already wanted to do this but I was just too lazy to pull through :)

File: Cargo.toml [hunk 1]
```diff
@@ -4,12 +4,15 @@ members = ["cargo-insta"]
 [workspace.metadata.dist]
 cargo-dist-version = "0.12.0"
 ci = ["github"]
+install-updater = false
 installers = ["shell", "powershell"]
-targets = ["aarch64-apple-darwin", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl", "x86_64-pc-windows-msvc"]
 pr-run-mode = "plan"
-install-updater = false
 precise-builds = true
+targets = ["aarch64-apple-darwin", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl", "x86_64-pc-windows-msvc"]
```
Comment (max-sixty) [inline Cargo.toml:11]: Bleh this was some autoformatting; can revert if annoying

**Comment by max-sixty** [comment]
This is ready for any feedback / testing. Will need to adjust the dependency spec if this gets in before we increase the `cargo-insta` msrv to 1.64...

**Comment by mitsuhiko** [comment]
Can you have a look at the changes to the CI matrix here?

**Comment by max-sixty** [comment]
OK, that should be fixed. We don't actually drop `syn` 1 yet, we need to bump MSRV further to get a newer `clap`

File: cargo-insta/Cargo.toml [hunk 2]
```diff
@@ -11,21 +11,24 @@ keywords = ["snapshot", "testing", "jest"]
 categories = ["development-tools::cargo-plugins"]
 edition = "2018"
 readme = "README.md"
-rust-version = "1.61.0"
+rust-version = "1.64.0"
```
Comment (max-sixty) [inline cargo-insta/Cargo.toml:14]: Note this bumps `cargo-insta` to 1.64, which was released ~2 years ago, Sep 2022

`insta` remains on `1.60.0`

**Pull request merged**